uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -655,27 +680,33 @@ define([
      * Loads time ranges from work pattern,
      * sets time ranges for timepickers and maximum value for deduction.
      *
-     * @param  {String} type (days|hours)
+     * @param  {String} dateType from|to
      * @return {Promise}
      */
-    function loadTimeRangesFromWorkPattern (type) {
-      var date = vm.uiOptions[type + 'Date'];
-      var timeObject = vm.uiOptions.times[type];
+    function loadTimeRangesFromWorkPattern (dateType) {
+      var date = vm.uiOptions[dateType + 'Date'];
+      var timeObject = vm.uiOptions.times[dateType];
       var isSingleDayRequest = !vm.uiOptions.multipleDays;
 
       if (!date) {
         return $q.resolve();
       }
 
       return vm.request.getWorkDayForDate(convertDateToServerFormat(date))
-        .then(function (response) {
-          timeObject.maxAmount = response.number_of_hours.toString() || '0';
+        .then(function (workDay) {
+          workDays[dateType] = workDay;
+
+          enableAndSetDataToTimeInput(dateType, workDay);
+          (isSingleDayRequest && dateType === 'from') && enableAndSetDataToTimeInput('to', workDay);
```

Comment:
I know it's a bit late and this is old, but why do we need to call `enableAndSetDataToTimeInput('to', workDay);` for single day requests? Aren't they hidden anyways?

Reply:
Nope, not hidden! :) https://github.com/civicrm/civihr/pull/2443